libretroshare/src/serialiser/rstypeserializer.cc

```diff
@@ -543,51 +543,56 @@ void RsTypeSerializer::RawMemoryWrapper::serial_process(
 		break;
 	case RsGenericSerializer::DESERIALIZE:
 	{
-        // In case first,second is not properly initialized, we set them to nullptr,0
-        first = nullptr;
-        second = 0;
-
-		uint32_t serialSize = 0;
-		RS_SERIAL_PROCESS(serialSize);
+		if(first || second)
+		{
+			/* Items are created anew before deserialization so buffer pointer
+			 * must be null and size 0 at this point */
+
+			RsErr() << __PRETTY_FUNCTION__ << " DESERIALIZE got uninitialized "
+			        << " or pre-allocated buffer! Buffer pointer: " << first
+			        << " must be null and size: " << second << " must be 0 at "
+			        << "this point. Does your item costructor initialize them "
+			        << "properly?" << std::endl;
+			print_stacktrace();
+			ctx.mOk = false;
+			break;
```

Comment:
actually we can afford to remove the break here. It's not a problem if that happens for what we do in this function. It's just a warning that will help us fixing badly initialized memory.

Reply:
done